lgc/patch/NggPrimShader.cpp

```diff
@@ -222,6 +224,21 @@ FunctionType *NggPrimShader::generatePrimShaderEntryPointType(uint64_t *inRegMas
     argTys.push_back(m_builder->getInt32Ty()); // Instance ID
   }
 
+  // If the ES is the API VS, and it is a fetchless VS, then we need to add args for the vertex fetches.
+  if (!hasTs) {
+    unsigned vertexFetchCount = m_pipelineState->getPalMetadata()->getVertexFetchCount();
+    if (vertexFetchCount != 0) {
+      // TODO: This probably will not work with culling.
+      if (m_hasGs)
```

Comment:
I think we only have to report error in this case for NGG: VS+FS (with culling). When we have GS, culling could be done regardless of whether you separate fetch shader from VS because the culling is based on GS output primitive attribute data (not VS fetch data).